test/unit/lachaise-tests.js

```
@@ -0,0 +1,253 @@
+
+                return cluster;
+            };
+            return self;
+        };
+
+        mockery.registerMock('couchbase', mockCouchbase());
+
+        var LaChaise = require('../../lib/lachaise.js');
+        var config = require('../mocks/mock-config.js');
+
+        var lac = new LaChaise(config);
+
+        lac.bucket('test', function(err, bucket) {});
+
+        bucketMock.connected = true;
+        bucketMock.emit('connect');
+
+        lac.bucket('test', function(err, bucket) {
+            expect(bucket).to.be.an.instanceOf(EventEmitter);
+            done();
+        });
+    });
+
+    it('should retry when it can\'t open a bucket', function(done) {
+        var spyOpenBucket = sinon.stub();
+        var mockOpenBucket = new EventEmitter;
+        mockOpenBucket.connected = true;
+
+        var mockCouchbase = function() {
+            var self = {};
+            self.Cluster = function() {
+                var cluster = {};
+
+                cluster.openBucket = spyOpenBucket;
+
+                return cluster;
+            };
+            return self;
+        };
+
+        function getConnectedBucket() {
+            mockOpenBucket.connected = true;
+            setTimeout(function() { mockOpenBucket.emit('connect'); }, 20);
+            return mockOpenBucket;
+        }
+
+        spyOpenBucket.onCall(0).returns(mockOpenBucket);
+        spyOpenBucket.onCall(1).returns(getConnectedBucket());
+
+        mockery.registerMock('couchbase', mockCouchbase());
+
+        var LaChaise = require('../../lib/lachaise.js');
+        var config = require('../mocks/mock-config.js');
+
+        var lac = new LaChaise(config);
+
+        lac.bucket('test', function() {
+            expect(spyOpenBucket.callCount).to.equal(2);
+            done();
+        });
+
+        mockOpenBucket.emit('error', { msg: 'retry-able', code: 16  });
+    });
+
+    it('should return on unrecoverable errors.', function(done) {
```

`it('should return an error on unrecoverable errors'...`